### PR TITLE
UI update

### PR DESCRIPTION
**Pull request by NateWr**
Ok, let's just do this thing. This represents a big re-organization of the frontend templates and LESS stylesheets to match the structure implemented in OJS 3.0b.

The frontend is really rough in terms of design. There's lots I'd like to do, but this should at least get the pieces in the right place so that we can all start focusing on fixing the backend while I'm working on the task mockup.

Got myself turned around with a rebase, so ended up with an extra merge commit on top. Hope that doesn't muck things up too bad.

### CODE REVIEW

File: classes/publicationFormat/PublicationFormat.inc.php [hunk 1]
```diff
@@ -517,6 +517,25 @@ function _checkRequiredFieldsAssigned() {
...
+			return '';
+		}
+
+		$dimensions = array(
+			$this->getWidth() . $this->getWidthUnitCode(),
+			$this->getHeight() . $this->getHeightUnitCode(),
+			$this->getThickness() . $this->getThicknessUnitCode(),
+		);
+
+		return join( __('notification.type.visitCatalog'), $dimensions );
```
Comment (asmecher) [inline classes/publicationFormat/PublicationFormat.inc.php]: OTOH that `join` should be removed.

Comment (asmecher) [inline classes/publicationFormat/PublicationFormat.inc.php]: Oh -- I see how it's being applied. I'd suggest moving the variables into the locale key itself, so they can be formatted as desired.

File: plugins/themes/default/styles/body.less [hunk 2]
```diff
@@ -0,0 +1,118 @@
+/**
+ * @file styles/structure/body.less
```
Comment (asmecher) [inline plugins/themes/default/styles/body.less]: Path needs updating.

File: plugins/themes/default/styles/components.less [hunk 3]
```diff
@@ -0,0 +1,218 @@
+/**
+ * @file styles/components.less
```
Comment (asmecher) [inline plugins/themes/default/styles/components.less]: Likewise (probably elsewhere too)

**Comment by asmecher** [comment]
Woo! That's a refreshing amount of pink to go with the green. About to check it out and give it a spin.

**Comment by asmecher** [comment]
OK, so it actually works (!) and it's great to see some of the new OJS layout stuff successfully brought in. Great stuff. I can see a lot of little things needing attention -- I'd probably start with the automated tests -- but I think it's probably worth merging now and getting cracking on master directly. Objections, @NateWr?

**Comment by NateWr** [comment]
Ok, pushed a couple of commits which should address your comments. I'd say merge when ready. Everything else should be piecemeal fixes and tweaks.

**Comment by asmecher** [comment]
Merged. Here we go!

**Pull request merged**